File: android/src/main/kotlin/com/frontegg/flutter/stateListener/FronteggStateListener.kt
package com.frontegg.flutter.stateListener

import io.flutter.plugin.common.EventChannel


interface FronteggStateListener {
    fun setEventSink(eventSink: EventChannel.EventSink?)
    fun subscribe()
    fun dispose()

    /** Emit the current auth state to Flutter on demand (used by `forceStateUpdate`, FR-25944). */
    fun forceNotifyChanges()
}

File: ios/frontegg_flutter/Sources/frontegg_flutter/stateListener/FronteggStateListener.swift
import Foundation
import Flutter

protocol FronteggStateListener {
    func setEventSink(eventSink: FlutterEventSink?)
    
    func subscribe()

    func dispose()

    /// Emit the current auth state to Flutter on demand (used by `forceStateUpdate`, FR-25944).
    func forceNotifyChanges()
}
